subscriptions-api.md

```diff
@@ -0,0 +1,613 @@
+event to receivers based on the information in the event context.
+
+#### Consumer
+
+An entity receiving and processing events. Consumers may receive events from
+producers directly or via intermediaries. A consumer might listen and wait for
+events to be delivered to it or it might actively solicit them from the producer
+or intermediary.
+
+#### Subscription
+
+A relationship established between a consumer and a producer or an intermediary.
+The subscription reflects the consumer's interest in receiving events and
+describes the method for how to deliver those events.
+
+#### Subscription Manager
+
+An entity, defined in this specification, that manages the lifecycle of a
+subscription on behalf of an event consumer and that distributes events to
+registered consumers.
+
+## 3. Event Subscriptions
+
+This specification defines mechanisms for CNCF CloudEvents event consumers to
+subscribe to events originating from producers on behalf of event sources. The
+software entity handling these subscriptions and responsible for distributing
+events is abstractly referred to as a "subscription manager".
+
+A compliant CloudEvents subscription manager MUST support at least one of the
+transport protocols referenced in this specification and it MUST implement the
+subscription mechanism referred to or defined in this specification.
+
+In some cases, the event producer MAY delegate the role of the subscription
+manager to some intermediary, in which case produced events are made available
+to the intermediary, and the intermediary's subscription manager determines who
+receives copies of the published events. In other cases, the event producer MAY
+take on the subscription manager role by itself.
+
+In this specification, we distinguish two styles of relationships between an
+event consumer and a subscription manager, whereby the key differentiator is how
+the event delivery channel to the consumer is initiated:
+
+- Subscriptions with consumer-solicited delivery ("pull"-style) are configured
+  on the subscription manager for delivering events through a communication
+  channel (like an AMQP link or a MQTT connection) initiated by the consumer
+  and connecting to the subscription manager. These kinds of subscriptions are
+  typically offered by messaging or eventing middleware and their lifetime
+  might be bounded by the lifetime of the communication channel.
+
+- Subscriptions with subscription-manager-initiated delivery ("push"-style) are
+  configured on the subscription manager for delivering events through a
+  communication channel that the subscription manager initiates when events
+  are available for delivery on the subscription. The configuration of such a
+  subscription MUST contain all information required for the subscription
+  manager to select a transport protocol, establish the desired communication
+  channel, and deliver the event(s).
+
+An application using CloudEvents might use only one of these styles or a
```

Comment:
the use of the word 'application' here might throw some people - would it be incorrect to switch it to "A consumer" ?

Reply:
And below...

Reply:
or use "An application acting as an event consumer..."

Reply:
hmmm having read the next paragraph, I think the use of the word "application" was very different than what I original thought. I think you mean something more akin to "the entire flow of an event from the source to the final consumer", right?

Reply:
Yes. I changed to this end-to-end solution.